### PR TITLE
feat(line-ripple): mdc-web typescript support

### PR DESCRIPTION
**Pull request by gugu**

### CODE REVIEW

File: test/unit/line-ripple/index.test.tsx [hunk 1]
```diff
@@ -3,10 +3,15 @@ import * as td from 'testdouble';
 import {assert} from 'chai';
 import {shallow} from 'enzyme';
 import LineRipple from '../../../packages/line-ripple/index';
+import {MDCLineRippleAdapter} from '@material/line-ripple/adapter';
 import {coerceForTesting} from '../helpers/types';
 
 suite('LineRipple');
 
+function getAdapter(foundation: any): MDCLineRippleAdapter {
```
Comment (moog16) [inline test/unit/line-ripple/index.test.tsx]: any --> MDCLineRippleFoundation

File: test/unit/line-ripple/index.test.tsx [hunk 2]
```diff
@@ -90,40 +95,42 @@ test('#adapter.hasClass returns true if exists in classList', () => {
...
 test('onTransitionEnd calls the #foundation.handleTransitionEnd', () => {
   const wrapper = shallow<LineRipple>(<LineRipple />);
-  wrapper.instance().foundation_.handleTransitionEnd = td.func();
-  const event = {test: '123'};
+  wrapper.instance().foundation_.handleTransitionEnd = td.func<(env: TransitionEvent) => null>();
+  const event: React.TransitionEvent = {
+    nativeEvent: {
+      test: '123',
+    },
+  } as any;
```
Comment (moog16) [inline test/unit/line-ripple/index.test.tsx]: please change to unknown to follow best practices

https://github.com/material-components/material-components-web/blob/master/docs/code/best_practices.md#any-vs-unknown-vs-

Comment (moog16) [reply test/unit/line-ripple/index.test.tsx]: Please use `coerceForTesting` https://github.com/material-components/material-components-web/blob/master/docs/code/best_practices.md#any-vs-unknown-vs-

File: test/unit/line-ripple/index.test.tsx (hunk 1 above)
Comment (gugu) [reply test/unit/line-ripple/index.test.tsx]: When I do this change, I get an error

`Argument of type 'import("/home/travis/build/material-components/material-components-web-react/packages/line-ripple/node_modules/@material/line-ripple/foundation").MDCLineRippleFoundation' is not assignable to parameter of type 'import("/home/travis/build/material-components/material-components-web-react/node_modules/@material/line-ripple/foundation").MDCLineRippleFoundation'.`

Looks like imports for testing and for the component are different. Do you know how to fix this error?

File: package.json [hunk 3]
```diff
@@ -71,7 +71,7 @@
     "@material/floating-label": "^0.41.0",
     "@material/icon-button": "^0.41.0",
     "@material/layout-grid": "^0.41.0",
-    "@material/line-ripple": "^0.41.0",
+    "@material/line-ripple": "^1.0.0-0",
```
Comment (moog16) [inline package.json]: update to 1.0.0

File: packages/line-ripple/package.json [hunk 4]
```diff
@@ -17,7 +17,7 @@
     "url": "https://github.com/material-components/material-components-web-react.git"
   },
   "dependencies": {
-    "@material/line-ripple": "^0.41.0",
+    "@material/line-ripple": "^1.0.0-0",
```
Comment (moog16) [inline packages/line-ripple/package.json]: 1.0.0

File: test/unit/line-ripple/index.test.tsx [hunk 5]
```diff
@@ -3,10 +3,16 @@ import * as td from 'testdouble';
 import {assert} from 'chai';
 import {shallow} from 'enzyme';
 import LineRipple from '../../../packages/line-ripple/index';
+import {MDCLineRippleAdapter} from '@material/line-ripple/adapter';
 import {coerceForTesting} from '../helpers/types';
 
 suite('LineRipple');
 
+function getAdapter(foundation: unknown): MDCLineRippleAdapter {
```
Comment (moog16) [inline test/unit/line-ripple/index.test.tsx]: RE
Argument of type 'import("/home/travis/build/material-components/material-components-web-react/packages/line-ripple/node_modules/@material/line-ripple/foundation").MDCLineRippleFoundation' is not assignable to parameter of type 'import("/home/travis/build/material-components/material-components-web-react/node_modules/@material/line-ripple/foundation").MDCLineRippleFoundation'.

Looks like imports for testing and for the component are different. Do you know how to fix this error?


It sounds like one of your deps are out of sync with each other. There are the node_modules for the @material/line-ripple package and the tests. Try deleting the node_modules directory of @material/line-ripple and the root directory, and running `npm i` again.

Comment (gugu) [reply test/unit/line-ripple/index.test.tsx]: the error is from Travis, it does not have any cache issues I think

Comment (moog16) [reply test/unit/line-ripple/index.test.tsx]: What if you do:
```

function getAdapter(instance: LineRipple): MDCLineRippleAdapter {
  // @ts-ignore
  return instance.foundation_.adapter_;
}```

Comment (gugu) [reply test/unit/line-ripple/index.test.tsx]: yes, that works, thanks

**Pull request merged**